### PR TITLE
Feature Get-vRARequestResult

### PR DESCRIPTION
**Pull request by otabut**
Enhancement
#165 - cmdlet for catching request result by invoking /catalog-service/api/consumer/requests/<id>/forms/details

Also need to modify PowervRA.psd1

### CODE REVIEW

File: src/Functions/Public/catalog-service/Get-vRARequestResult.ps1 [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    .PARAMETER RequestNumber
+    The request number of the request to query
+
+    .INPUTS
+    System.String
+
+    .OUTPUTS
+    System.Management.Automation.PSObject
+
+    .EXAMPLE
```
Comment (chelnak) [inline src/Functions/Public/catalog-service/Get-vRARequestResult.ps1]: Add an example to show pipeline support:

```
Get-vRARequest -RequestNumber 339 | Get-vRARequestResult
```

File: src/Functions/Public/catalog-service/Get-vRARequestResult.ps1 [hunk 2]
```diff
@@ -0,0 +1,94 @@
+
+function Get-vRARequestResult {
```
Comment (chelnak) [inline src/Functions/Public/catalog-service/Get-vRARequestResult.ps1]: Possible function rename? I feel that Get-vRARequestDetail would suit better as Get-vRARequest actually returns the result of the request.. @jonathanmedd Interested in your views

File: src/Functions/Public/catalog-service/Get-vRARequestResult.ps1 [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+                        break
+
+                }
+                # --- If the request number parameter is passed returned detailed information about the request
+                'ByRequestNumber' {
+
+                        $id = (Get-vRARequest -RequestNumber $RequestNumber).id
+
+                        Get-vRARequestResult -id $id
```
Comment (chelnak) [inline src/Functions/Public/catalog-service/Get-vRARequestResult.ps1]: Avoid this recursion if possible... just use the same pattern that you did in the 'ById' block 👍

**Comment by otabut** [comment]
Sorry for wrong actions on GitHub, I'm pretty new in this exercice :p
Feel free to discard any incorrect operation

File: src/Functions/Public/catalog-service/Get-vRARequestDetail.ps1 [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+                        break
+
+                }
+                # --- If the request number parameter is passed returned detailed information about the request
+                'ByRequestNumber' {
+
+                        $id = (Get-vRARequest -RequestNumber $RequestNumber).id
+
+                        Get-vRARequestDetail -id $id
```
Comment (chelnak) [inline src/Functions/Public/catalog-service/Get-vRARequestDetail.ps1]: Just this one left to change now.. i would do something similar to this (would need testing though):

```
$Id = (Get-vRARequest -RequestNumber $RequestNumber).id
$URI = "/catalog-service/api/consumer/requests/$($Id)/forms/details"
$RequestDetail = Invoke-vRARestMethod -Method GET -URI $URI -Verbose:$VerbosePreference
$RequestDetail.values.entries
```

**Comment by chelnak** [comment]
@otabut No need to apologise! We appreciate your contribution. There's just one more change to make, which i've mentioned it above, then @jonathanmedd will then review it. 

This will go in to the next release!

File: src/Functions/Public/catalog-service/Get-vRARequestDetail.ps1 [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+    .EXAMPLE
+    Get-vRARequestDetail -Id 972ab103-950a-4240-8a3d-97174ee07f35
+    
+    .EXAMPLE
+    Get-vRARequestDetail -RequestNumber 965299
+    
+    .EXAMPLE
+    Get-vRARequest -RequestNumber 965299 | Get-vRARequestDetail
+
```
Comment (jonathanmedd) [inline src/Functions/Public/catalog-service/Get-vRARequestDetail.ps1]: Can you also add an example for supporting multiple Ids or numbers, e.g.

```
.EXAMPLE
Get-vRARequestDetail -RequestNumber 965299,965300
```

**Pull request merged**